### PR TITLE
Add kpack-dependencies package

### LINKED ISSUE #3668: Proposal to add package for kpack dependencies
## Feature Request

Currently, the kpack package in TCE requires a lot of manual steps to help bootstrap its installation with dependencies. This results in a poor install experience for users. We would like to create a new package that will provide those dependencies for users. The dependencies package would be updated at some cadence by a pipeline, pulling in the latest paketo stacks and buildpacks. Users would just need to provide an repository location and secret for the resources to be created.

## Describe alternatives you've considered

Otehr alternatives we have considered involve creating a "descriptor file" that would be used by the [kpack cli](https://github.com/vmware-tanzu/kpack-cli/blob/main/docs/kp_import.md) to import dependencies. This would require manual user intervention as well as a way for us to distribute the files to users.

## Additional context

This will be used by the App Toolkit package to provide a better experience for install

### PR DESCRIPTION
**Pull request by tylerphelan**
## What this PR does / why we need it

Adding the kpack dependencies package to TCE

## Which issue(s) this PR fixes

Fixes: https://github.com/vmware-tanzu/community-edition/issues/3668

## Describe testing done for PR

Integration testing with kpack package.
Test coverage includes builders and image creation functionality.

### Cadence

Releasing this package is low cost for the kpack team, so we are happy to release on any cadence. The only blocker for release cadence is that it is not currently a fully automated process so release is gated on creating the PR manually which could be a blocker for continuous delivery.

### Important for Docs

There are 3 important things users must know so we tried to make them clear in the docs:
1. The kpack package is a hard dependency and it must be installed first.
2. To install kpack-dependencies package the user only needs to provide the `kp_default_repository`, however it MUST be the same value used to install the kpack package.
  - This is because the authentication credentials used to install kpack are the same credentials used to write to `kp_default_repository` in the kpack-dependencies package.
3. The resources that are created by the kpack dependencies package cannot be modified.
  - This is because when deployed by kapp controller, resources lifecycle is managed entirely by kapp controller and any user changes would be overwritten.
  - This is in contrast to other kpack workflows that recommend managing and modifying existing ClusterStore, ClusterStack, and ClusterBuilder resources.
  - It should be made clear that if users want custom solutions they will need to create and manage them manually.
4. The installation does not "fail fast" and will reconcile until the timeout if the installation is in a failing state.
  - This is due to kpack implementation and can likely be improved in later releases.

### Notes

- The buildpacks and stacks for each release will be available at the package-for-kpack-dependencies github release page: https://github.com/vmware-tanzu/package-for-kpack-dependencies/releases/tag/v0.0.5
  - At the time of this PR the https://github.com/vmware-tanzu/package-for-kpack-dependencies page is not yet approved for being converted to a public repo but is in progress.
- Initial release version 0.0.5, this is an arbitrary version number.
- Bundle is distributed from projects.registry.vmware.com/kpack/dependencies, not projects.registry.vmware.com/tce/kpack-dependencies due to issues with immutable tags.

### CODE REVIEW

**Comment by kartiklunkad26** [comment]
@seemiller for review!

File: addons/packages/kpack-dependencies/0.0.5/README.md [hunk 1]
```diff
@@ -0,0 +1,453 @@
...
+base      <some-image>   True
+default   <some-image>   True
+```
+
+## Configuration and Usage
+
+After installing kpack dependencies dependencies do not need to be manually installed, and you can immediately create source-to-image builds.
+
+* [Creating an image using kp](#creating-an-image-using-kp-cli)
+* [Creating an image using kubectl](#creating-an-image-using-kube 6ctl)
```
Comment (seemiller) [inline addons/packages/kpack-dependencies/0.0.5/README.md:71]: @tylerphelan I think there is a typo here - `kube 6ctl`,  and the link is not rendering correctly.
```
[Creating an image using kubectl](#creating-an-image-using-kube 6ctl)
```

**Review by seemiller — CHANGES_REQUESTED** [review]

**Review by seemiller — DISMISSED** [review]

**Comment by seemiller** [comment]
@tylerphelan @kartiklunkad26 this PR is pulling in 0.0.5, but I see that 0.0.9 is available, would you want to bump the PR to use the latest version of the package? (Latest version also has the readme fixes I asked for)

**Comment by tylerphelan** [comment]
updated to 0.0.9 latest kpack dependencies release

**Comment by seemiller** [comment]
Created a package repo for testing - `projects-stg.registry.vmware.com/tce/main@sha256:9da3352a8702e5f6ecf8160d26dc5beb59430501e980677ac521b43c932653ac`

File: addons/packages/kpack-dependencies/0.0.9/README.md [hunk 2]
```diff
@@ -0,0 +1,453 @@
...
+
+$ kubectl get clusterbuilder
+NAME      LATESTIMAGE    READY
+base      <some-image>   True
+default   <some-image>   True
+```
+
+## Configuration and Usage
+
+After installing kpack dependencies dependencies do not need to be manually installed, and you can immediately create source-to-image builds.
```
Comment (kcoriordan) [inline addons/packages/kpack-dependencies/0.0.9/README.md:68]:
```suggestion
After installing kpack, dependencies do not need to be manually installed, and you can immediately create source-to-image builds.
```

File: addons/packages/kpack-dependencies/0.0.9/README.md [hunk 3]
```diff
@@ -0,0 +1,453 @@
...
+     kubectl apply -f service-account.yaml
+     ```
+
+3. Create a kpack Image Resource.
+
+   An Image Resource is the specification for an OCI image that kpack should
+   build and manage.
+
+   We will create a sample Image Resource that builds with the builder created
+   in step #5.
```
Comment (kcoriordan) [inline addons/packages/kpack-dependencies/0.0.9/README.md:299]: should this be step 4?

File: addons/packages/kpack-dependencies/0.0.9/README.md [hunk 4]
```diff
@@ -0,0 +1,453 @@
...
+   ```
+
+4. Rebuilding kpack Image Resources.
+
+   We recommend updating the kpack Image Resource with a CI/CD tool when new
+   commits are ready to be built.
+   > Note: You can also provide a branch or tag as the `spec.git.revision` and kpack will poll and rebuild on updates!
+
+   We can simulate an update from a CI/CD tool by updating
+   the `spec.git.revision` on the Image Resource configured in step #6.
```
Comment (kcoriordan) [inline addons/packages/kpack-dependencies/0.0.9/README.md:206]: Should this refer to step 3 --- Create a kpack Image Resource.

File: addons/packages/kpack-dependencies/0.0.9/README.md [hunk 5]
```diff
@@ -0,0 +1,453 @@
...
+
+   You should see a new build with
+
+   ```text
+   BUILD    STATUS     IMAGE                                            REASON
+   1        SUCCESS    index.docker.io/your-name/app@sha256:6744b...    BUILDPACK
+   2        BUILDING                                                    CONFIG
+   ```
+
+   You can tail the logs for the Image Resource with the kp cli used in step #6.
```
Comment (kcoriordan) [inline addons/packages/kpack-dependencies/0.0.9/README.md:232]: should this be step 2?

**Comment by tylerphelan** [comment]
@kcoriordan resolved by removing all references to other steps as they are unnecessary

**Review by kcoriordan — DISMISSED** [review]

**Comment by tylerphelan** [comment]
@seemiller @kcoriordan 
added some context for troubleshooting the installation in the README due to:

>The installation does not "fail fast" and will reconcile until the timeout if the installation is in a failing state.

**Review by kcoriordan — APPROVED** [review]

**Review by seemiller — APPROVED** [review]

**Pull request merged**